modules/base/include/modules/base/algorithm/camerautils.h

```diff
@@ -0,0 +1,190 @@
+                                       bool setLookRanges = false);
+
+/**
+ * Set the ranges of the look to and look from properties of the camera. Will center around the mid
+ * point of dataToWorld. The lookTo will be set to ranges to stay within the volume. The lookFrom
+ * will be set to the mid point +- the basis vector of dataToWorld times the zoom factor. That is a zoom
+ * factor of 25 will allow to zoom out to a distance of "25 volumes".
+ *
+ * @param cam           camera to update
+ * @param dataToWorld   bounding volume used to determinte the ranges
+ * @param maxZoomFactor determines how far away from the volume the user will be able to zoom out.
+ */
+IVW_MODULE_BASE_API void setCameraLookRanges(CameraProperty &cam, const mat4 &dataToWorld,
+                                             float maxZoomFactor = 25.f);
+
+/**
+ * Computes appropriate near and far clip distances for the given bounding volume and zoom factor. Makes
+ * sure that the far plane is distant enough to avoid clipping given to current zoomfactor.
+ * @see setCameraLookRanges
+ */
+IVW_MODULE_BASE_API std::pair<float, float> computeCameraNearFar(
+    const mat4 &dataToWorld, float maxZoomFactor = 25.f, float nearFarRatio = 1.f / 10000.f);
+
+/**
+ * Sets the near and far clip distances of the camera based on the given bounding volume and zoom
+ * factor. Ensures that the far plane is distant enough to avoid clipping given to current zoomfactor.
+ * @see computeCameraNearFar
+ * @see setCameraLookRanges
+ */
+IVW_MODULE_BASE_API void setCameraNearFar(CameraProperty &cam, const mat4 &dataToWorld,
+                                          float maxZoomFactor = 25.f,
+                                          float nearFarRatio = 1.f / 10000.f);
+
+/**
+ * Helper class for wrapping properties and setting up their callbacks to update the given camera.
+ * Can be used together with mesh ports (both single, multi and flat) and volume ports.
+ * @see setCameraView
+ * @see setCameraLookRanges
+ * @see setCameraNearFar
+ */
+class IVW_MODULE_BASE_API FitCameraPropertiesHelper {
+    FitCameraPropertiesHelper(std::string identifier, std::string displayName);
+
+public:
+    FitCameraPropertiesHelper(std::string identifier, std::string displayName,
+                              CameraProperty &camera, VolumeInport &volumePort);
+
+    template <size_t N, bool Flat>
+    FitCameraPropertiesHelper(std::string identifier, std::string displayName,
+                              CameraProperty &camera, DataInport<Mesh, N, Flat> &meshPort);
+
+    CompositeProperty &getCompositeProperty() { return composite_; }
+
+private:
+    CompositeProperty composite_;
+    CompositeProperty lookAt_;
+    CompositeProperty lookAtSettings_;
+    ButtonProperty flipUp_;
+    BoolProperty updateNearFar_;
+    BoolProperty updateLookRanges_;
+    FloatProperty fittingRatio_;
+    ButtonProperty xNegative_;
```

Comment:
Suggest to replace the buttons with an option property with the enum options along with a synced vec3 property for camera direction

Reply:
No. I want the buttons to click on. Would just be very annoying having the change the option property all the time when trying to find a nice view.

Reply:
Buttons are more intuitive. Similar, assume the option property is set to "X+", then the camera position is changed. What do you do to reset the camera again to "X+"? With the button it's a single click, using the option property you need to switch to a different view direction first(?).

Reply:
I guess it would in its current state need an apply button as well. Less UI space consumed but more interaction. I guess it is more a matter of taste...

Reply:
Maybe I'm spoiled by other software ;-)

Reply:
A different property semantic with condensed button layout for option property had probably been optimal. Or a better packed property group